Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectAdapterFactory.java

```diff
@@ -43,12 +43,20 @@
     private static final String BEAN_NAME_MANAGE_TASK_SCHEDULER = "manageTaskScheduler";
 
     /**
-     * Register Handlers will invoke getInstance, thereby loading the spring
-     * context and task scheduler for polling mail servers.
+     * Register Handlers will invoke getInstance, thereby loading the spring context and task scheduler for polling mail
+     * servers.
      */
     public void registerHandlers() {
-        //initialize the HibernateUtil when the Direct Servlet is initialized.. DO NOT Remove this.
-        SessionFactory session = gov.hhs.fha.nhinc.event.persistence.HibernateUtil.getSessionFactory();
+        /**
+         * TODO: There is a race condition on accessing the SessionFactory for both Direct Message Monitoring a Event
+         * Persistence; initializing both when the Direct Servlet is initialized as a workaround.
+         * <br/>
+         * DO NOT remove either of the following two lines of code until this issue is resolved.
+         */
+        SessionFactory eventSession = gov.hhs.fha.nhinc.event.persistence.HibernateUtil.getSessionFactory();
+        SessionFactory msgMonitorSession = gov.hhs.fha.nhinc.direct.messagemonitoring.persistence.HibernateUtil
```

Comment:
There are two "HibernateUtil" classes used back-to-back, and we can only import one (at most) anyway.  I think it would be more confusing to import just one, rather than to fully qualify both.